### PR TITLE
Update Python versions.

### PR DESCRIPTION
**Pull request by jugmac00**
- Drop support for Python 3.4.

- Add support for Python 3.8-pre.

modified:   .travis.yml
modified:   CHANGES.rst
modified:   setup.py
modified:   tox.ini

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -58,10 +58,10 @@ def read(*rnames):
         'Programming Language :: Python :: 2',
         'Programming Language :: Python :: 2.7',
         'Programming Language :: Python :: 3',
-        'Programming Language :: Python :: 3.4',
         'Programming Language :: Python :: 3.5',
         'Programming Language :: Python :: 3.6',
         'Programming Language :: Python :: 3.7',
+        'Programming Language :: Python :: 3.8-pre',
```
Comment (mgedmin) [inline setup.py]: I do not see 3.8-pre on https://pypi.org/classifiers/

Comment (jugmac00) [reply setup.py]: Thank you for the link. Just updated the classifier to 3.8 (though it seems a bit odd, as 3.8 final is not yet out).

**Comment by mgedmin** [comment]
I will clarify my position: it's a good idea to keep master always in a releasable state, in case someone needs to make an urgent bugfix release.  twine upload will fail if setup.py has a classifier not on the approved list, which would introduce unnecessary friction into the release process.

**Comment by jugmac00** [comment]
Thank you for the explanation. There is much to learn, and detailed explanations make it so much easier to catch up.

**Pull request merged**